### PR TITLE
Port sponsorship tracking

### PR DESCRIPTION
**Pull request by ecino**
Wait to have module sponsorship_compassion before merging, needs to be tested.

### CODE REVIEW

File: sponsorship_tracking/model/contracts.py [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+        We also put the person who terminated the contract as follower. """
+        res = super(recurring_contract, self).contract_terminated()
+        self.write({'project_state': False,
+                    'sds_uid': self.env.user.id})
+        return res
+
+    ##########################################################################
+    #                             PRIVATE METHODS                            #
+    ##########################################################################
+    def _define_parent_id(self, correspondant_id):
```
Comment (steveferry) [inline sponsorship_tracking/model/contracts.py]: `@api.multi` ?

File: sponsorship_tracking/model/contracts.py [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+        same_partner_contracts = self.search(
+            [('correspondant_id', '=', correspondant_id),
+             ('sds_state', '=', 'sub_waiting')])
+        for same_partner_contract in same_partner_contracts:
+            if not self.search_count([
+                    ('parent_id', '=', same_partner_contract.id)]):
+                return same_partner_contract.id
+        return False
+
+    def _parent_id_changed(self, parent_id):
```
Comment (steveferry) [inline sponsorship_tracking/model/contracts.py]: `@api.multi` ?

File: sponsorship_tracking/model/contracts.py (hunk 1 above)
Comment (ecino) [inline sponsorship_tracking/model/contracts.py]: Not necessary in private methods (only called from this class, api.multi is by default)

File: sponsorship_tracking/model/contracts.py (hunk 2 above)
Comment (ecino) [inline sponsorship_tracking/model/contracts.py]: Not necessary

**Comment by ecino** [comment]
Pull request updated with code corrections and use of new workflow APIs. @steveferry Can you do a review please ?

File: sponsorship_compassion/model/contracts.py [hunk 3]
```diff
@@ -334,11 +334,7 @@ def _suspend_change_invoices(self, since_date, product):
...
         cancel_inv_lines.write(vals)
-
-        wf_service = netsvc.LocalService('workflow')
-        for invoice_id in invoices.ids:
-            wf_service.trg_validate(self.env.user.id, 'account.invoice',
-                                    invoice_id, 'invoice_open', self.env.cr)
+        invoices.signal_workflow('invoice_open')
 
     @api.multi
     def get_suspend_invl_data(self, product):
```
Comment (steveferry) [inline sponsorship_compassion/model/contracts.py]: Rename `product` in `product_id` and use it directly in the dictionnary

File: sponsorship_tracking/model/contracts.py [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+            'view_mode': 'form',
+            'views': [(view_id, 'form')],
+            'res_model': self._name,
+            'type': 'ir.actions.act_window',
+            'target': 'current',
+            "res_id": self.ids[0],
+        }
+
+    @api.multi
+    def mail_sent(self):
```
Comment (steveferry) [inline sponsorship_tracking/model/contracts.py]: Is it really useful?

File: sponsorship_tracking/model/contracts.py [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+        return self.signal_workflow('mail_sent')
+
+    @api.multi
+    def project_mail_sent(self):
```
Comment (steveferry) [inline sponsorship_tracking/model/contracts.py]: Is it really useful?

File: sponsorship_tracking/wizard/sub_sponsorship_wizard.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        if not child:
+            raise exceptions.Warning(
+                _("No child selected"),
+                _("Please select a child"))
+
+        sponsorship_id = self.env.context.get('active_id')
+        contract_obj = self.env['recurring.contract']
+        contract = contract_obj.browse(sponsorship_id)
+        origin_obj = self.env['recurring.contract.origin']
+        sub_origin_ids = origin_obj.search([('type', '=', 'sub')]).ids
```
Comment (steveferry) [inline sponsorship_tracking/wizard/sub_sponsorship_wizard.py]: `sub_origin_id = self.env['recurring.contract.origin'].search([('type', '=', 'sub')], limit=1)`

File: sponsorship_tracking/model/contracts.py (hunk 4 above)
Comment (ecino) [inline sponsorship_tracking/model/contracts.py]: Yes as it is a method called by a button from the view.

File: sponsorship_tracking/model/contracts.py (hunk 5 above)
Comment (ecino) [inline sponsorship_tracking/model/contracts.py]: Button from the view.

**Comment by ecino** [comment]
Code corrected.

**Comment by steveferry** [comment]
:+1:

**Pull request merged**